src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -332,106 +342,122 @@ internal void PerformBackup(BackupOperation backupOperation)
         {
             backupOperation.PerformBackup();
         }
-        
+
         /// <summary>
-        /// Create a backup task for execution and cancellation
+        /// Create a backup script task
         /// </summary>
         /// <param name="sqlTask"></param>
         /// <returns></returns>
-        internal async Task<TaskResult> BackupTaskAsync(SqlTask sqlTask)
+        internal async Task<TaskResult> BackupScriptTaskAsync(SqlTask sqlTask)
         {
-            sqlTask.AddMessage(SR.Task_InProgress, SqlTaskStatus.InProgress, true);
+            sqlTask.AddMessage(SR.TaskInProgress, SqlTaskStatus.InProgress, true);
             IBackupOperation backupOperation = sqlTask.TaskMetadata.Data as IBackupOperation;
-            TaskResult taskResult = null;
+            TaskResult taskResult = new TaskResult();
+            string script = "";
 
             if (backupOperation != null)
             {
-                AutoResetEvent backupCompletedEvent = new AutoResetEvent(initialState: false);
-                Task<TaskResult> performTask = PerformTaskAsync(backupOperation);
-                Task<TaskResult> cancelTask = CancelTaskAsync(backupOperation, sqlTask.TokenSource.Token, backupCompletedEvent);
-                Task<TaskResult> completedTask = await Task.WhenAny(performTask, cancelTask);
-
-                // Release the cancelTask
-                if (completedTask == performTask)
+                await Task.Factory.StartNew(() =>
                 {
-                    backupCompletedEvent.Set();
-                }
-
-                sqlTask.AddMessage(completedTask.Result.TaskStatus == SqlTaskStatus.Failed ? completedTask.Result.ErrorMessage : SR.Task_Completed,
-                                   completedTask.Result.TaskStatus);
-                taskResult = completedTask.Result;
+                    try
+                    {
+                        script = backupOperation.ScriptBackup();
+                        taskResult.TaskStatus = SqlTaskStatus.Succeeded;
+                        sqlTask.AddScript(taskResult.TaskStatus, script);
```

Comment:
This should be separated from task completion. As mentioned, I'd recommend:
- Separate messages for script generation vs. task completion
- Separate messages for task execution start / in progress / complete vs. task completion.